.github/workflows/r.yml

```diff
@@ -234,8 +234,8 @@ jobs:
       fail-fast: false
       matrix:
         config:
-        - { rtools: 42, rversion: "4.2" }
```

Comment:
Do we still need to test on rtools 42? I don't know the difference between 42 and 43 to know if not having any jobs that use it is a risk.

Reply:
>From the user perspective, Rtools43 is the same as Rtools42. It uses newer versions of the compiler toolchain and libraries, and hence some package authors will have to extend their make files to link additional libraries. Maintainers of CRAN and Bioconductor packages may use [these patches](https://www.r-project.org/nosvn/winutf8/ucrt3/patches/) for reference or re-use them in their code.

Does not look like there are any changes that affect us, so we can probably just update to 43?

Reply:
Different compiler version is potentially concerning, but the question is if there is enough risk to justify another hour-long build. Probably not?

What about nightly packages?

Reply:
> What about nightly packages?

How do you mean?

Reply:
I just  tried to update the version used in nightly packages and that's a no go for now (at least for the 10.13 runner...) as rig doesn't find 4.3 due to url change and even compiling from source I was unable to make any newer rig versions run on 10.13. We could do symlinking by hand or something but I'd rather not... hopefully we will be able to drop 10.13 soon :crossed_fingers: 

Also another thing regarding rtools: do we want to continure building urct + 32/64 bit versions of libarrow? imo it would be sensible to drop at the minimum the 32 bit build... but we don't really have a support policy so :shrug: (something I plan to bring up on the ml again)

Reply:
Well then people who want to use arrow on old versions on windows can use RSPM? I am strongly +1 on dropping 32bit support. Win 7 has been eol for 3 years and I don't think it is our responsibility to support such outdated configurations. I would also assume that the percentage of users that use this is rather small? So not worth it from that direction either (compare to java 8 discussion on the ml)

Reply:
RSPM (technically, PPPM today) will need the libarrow binaries to build for those older R versions, so it's not a solution to the problem, it's actually a reason to keep building the 32-bit packages.

Reply:
So (as we are following the past-4 rule I guess?) we will drop the 32bit build when 4.6 is released (aka in 3! years)?  My only reason to drop it is shortening of build times, as we build 3 libarrow versions in each windows build so :shrug:

Reply:
Valid concern, though in regular CI we only build the ucrt64 version, the 32-bit is only nightly, so the build time is less of an issue there. (I think the libarrow builds are also parallelized.)

That said, we already "violate" the tidyverse standard for version compatibility (Windows only) because we require R 4.0 on Windows due to C++17. 

I think I'm landing in the same place as @paleolimbot here: leave it in unless/until the 32-bit builds are causing undue developer burden, which it seems like today they are not.